rehttp_server_post17_test.go

```diff
@@ -0,0 +1,167 @@
+//go:build go1.7
+// +build go1.7
+
+package rehttp
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestTransport_RoundTripTimeouts(t *testing.T) {
+	// to keep track of any open server requests and ensure the correct number of requests were made
+	ch := make(chan bool, 4)
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ch <- true
```

Comment:
The problem with this approach is that the test could block forever (well, until `go test` times out) if the assumptions are not met, instead of failing fast (i.e. if the client happens to make 5 requests, the 5th one would block forever because the channel is full and no receives have been done yet).

I have an idea on how to make the test more robust, I'll try to take a shot at it when I have a moment, the general idea would be to spin a goroutine that receives in a `for range` on the channel, and close the channel after the call to `client.Do` so that the goroutine exits, and unblocks another channel (or a WaitGroup, but the nice thing with a channel is that the test could do a `select` on that channel and a timer so that it fails fast if for some reason it takes longer than expected to run the test).

That's not super easy to explain, that's why I say I'll take a shot at implementing it when I have a moment.

Reply:
I'm having a hard time understanding exactly what you're describing, but I agree that if this does fail it ends up in the undesirable state of blocking on the channel until the test times out. I don't have a great idea for solving this, but will take a look at your change set to hopefully learn something when you add it in.

Thanks for spending the time to read, review, and re-review this.